client/src/search.tsx

```diff
@@ -175,76 +200,76 @@ export class SearchWidgetClass extends React.Component {
-
-        if (q === "/") {
-          this.setState({
-            highlitResult: null,
-            searchResults: [],
-            showSearchResults: true,
-          });
-        } else {
-          const fuzzyResults = this.fuzzySearcher.search(q, { limit });
-          const results = fuzzyResults.map((fuzzyResult) => {
-            return {
-              title: this._map[fuzzyResult.needle].title,
-              uri: fuzzyResult.needle,
-              substrings: fuzzyResult.substrings,
-            };
-          });
-          this.setState({
-            highlitResult: results.length ? 0 : null,
-            searchResults: results,
-            showSearchResults: true,
-          });
-        }
-      } else {
-        // Full-Text search
-        const indexResults = this.index.search(q, {
-          limit,
-          // bool: "or",
-          suggest: true, // This can give terrible result suggestions
-        });
+    }
 
-        const results = indexResults.map((uri) => {
-          return {
-            title: this._map[uri].title,
-            uri,
-            popularity: this._map[uri].popularity,
-          };
-        });
+    // The iPhone X series is 812px high.
+    // If the window isn't very high, show fewer matches so that the
+    // overlaying search results don't trigger a scroll.
+    const limit = window.innerHeight < 850 ? 5 : 10;
+
+    let results = null;
+    if (value.startsWith("/") && !/\s/.test(value)) {
+      // Fuzzy-String search on the URI
+
+      if (value === "/") {
         this.setState({
-          highlitResult: results.length ? 0 : null,
-          searchResults: results,
+          highlitResult: null,
+          searchResults: [],
           showSearchResults: true,
         });
+      } else {
+        const fuzzyResults = this.fuzzySearcher.search(value, { limit });
+        results = fuzzyResults.map((fuzzyResult) => ({
+          title: this._map[fuzzyResult.needle].title,
+          uri: fuzzyResult.needle,
+          substrings: fuzzyResult.substrings,
+        }));
       }
+    } else {
+      // Full-Text search
+      const indexResults = this.index.search(value, {
+        limit,
+        // bool: "or",
+        suggest: true, // This can give terrible result suggestions
+      });
+
+      results = indexResults.map((uri) => ({
+        title: this._map[uri].title,
+        uri,
+        popularity: this._map[uri].popularity,
+      }));
+    }
+
+    if (results) {
+      onValueExistsChange &&
+        onValueExistsChange(
+          results.some((item) => item.uri.toLowerCase() === value.toLowerCase())
+        );
+      this.setState({
+        highlitResult: results.length ? 0 : null,
+        searchResults: results,
+        showSearchResults: true,
+      });
+    } else {
+      onValueExistsChange && onValueExistsChange(false);
     }
   };
 
   keyDownHandler = (event) => {
-    if (event.key === "Escape") {
+    if (event.key === "Enter") {
+      this.submitHandler();
```

Comment:
had to remove the form element wrapping the search, to make it usable in the document form (form nesting is forbidden in HTML). Hence this is now explicit.